### DISCUSSION TITLE
'Pio' was not declared in this scope

### DISCUSSION THREAD

**Discussion by gallaugher** [Makuna/NeoPixelBus#470] [Q&A] [answered]
Am using the library as part of Jiri Praus's Beating LED Heart code, but am running it on an Adafruit Feather nrf52840:
I get the following error:
"'Pio' was not declared in this scope"
referring to:
NeoArmMethod.h:765:9

Jiri's code is at:
https://content.instructables.com/ORIG/F37/A9BV/K8CG6VDT/F37A9BVK8CG6VDT.ino

So sorry if I've missed something obvious. Still fairly new to Arduino.  His include order in the main code is:
#include <NeoPixelBus.h>
#include <NeoPixelAnimator.h>
#include <Wire.h>
#include "MAX30105.h"           //MAX3010x library
#include "heartRate.h"          //Heart rate calculating algorithm

Struggling to nail this last bit for a Mother's Day gift. Thanks!

**Comment by Makuna** [comment, marked as answer]
This is usually due to the build not setting the Arduino Arch symbol correctly; and ARM support are the worse and using PIO will make it worse.

ARDUINO_ARCH_NRF52840 must be defined.  Also "underscore underscore" arm "underscore underscore" must also be set if you do want the bitbang.

With this you get the hardware support for this specific chip which is async.

**Reply by gallaugher** [reply]
Thanks - I wasn't able to get the nrf52840 to work, but it worked fine with the Feather M0. Thanks for your work! Send you a few bucks for a beer or other beverage of your choice. Not much given how exceptional your work has been, but I really appreciate your kindness & skill. Wishing you the best!
